### PR TITLE
[Merged by Bors] - Update `mouse.rs` docs in `bevy_input`

### PR DESCRIPTION
**Pull request by KDecay**
# Objective

- Part of the splitting process of #3692.

## Solution

- Document `mouse.rs` inside of `bevy_input`.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_input/src/mouse.rs
Comment (nicopap) [inline crates/bevy_input/src/mouse.rs:28]: I don't understand this bit: "The resource stores the data of the buttons of a mouse"
```suggestion
/// resource. It can be accessed inside of a system with a `Res<Input<MouseButton>>`
/// parameter.
```
(Second half is just a suggestion. It's only the "data of a button" that I think should be really fixed)

File: crates/bevy_input/src/mouse.rs
Comment (nicopap) [inline crates/bevy_input/src/mouse.rs:56]: Nit:
```suggestion
/// The scroll unit.
///
/// Describes how a value of a [`MouseWheel`](crate::mouse::MouseWheel) event has to be interpreted.
```

File: crates/bevy_input/src/mouse.rs
Comment (nicopap) [inline crates/bevy_input/src/mouse.rs:93]: :+1: Great to link to the relevant doc.

File: crates/bevy_input/src/mouse.rs
Comment (nicopap) [inline crates/bevy_input/src/mouse.rs:77]:
```suggestion
```

File: crates/bevy_input/src/mouse.rs
Comment (nicopap) [inline crates/bevy_input/src/mouse.rs:49]:
```suggestion
```

File: crates/bevy_input/src/mouse.rs
Comment (nicopap) [inline crates/bevy_input/src/mouse.rs:8]:
```suggestion
```

Is it relevant to the end user to say "this comes from the `winit` crate"? I don't feel like so.

Regardless, I think "It is available to the end user and can be used for game logic." doesn't add much value and should be removed.

**Review by nicopap — CHANGES_REQUESTED** [review]
Finally some doc! :+1:

Though there are some awkward bits that I think could be left out.

File: crates/bevy_input/src/mouse.rs
Comment (Sheepyhead) [inline crates/bevy_input/src/mouse.rs:42]:
```suggestion
    /// Another mouse button with the associated number.
```

File: crates/bevy_input/src/mouse.rs
Comment (Sheepyhead) [inline crates/bevy_input/src/mouse.rs:52]:
```suggestion
    /// The delta of the previous and current mouse positions.
```

**Review by Sheepyhead — CHANGES_REQUESTED** [review]

**Comment by KDecay** [comment]
> I don't understand this bit: "The resource stores the data of the buttons of a mouse"

@nicopap I can see how it could be confusing. Do you like "The resource stores the state of the mouse buttons" more?

> Is it relevant to the end user to say "this comes from the winit crate"? I don't feel like so. Regardless, I think "It is available to the end user and can be used for game logic." doesn't add much value and should be removed.

I wouldn't say so. The `winit` information might be a little much, but I think it doesn't hurt to leave it in there. The sentence that it is available to the end user and that it can be used for game logic seems like an important notice if you are just skimming through the input docs. 

The addition of the actual resource type you would have to use (`Res<Input<MouseButton>>`) was something I had inside of there. I removed it, because @alice-i-cecile noticed that it is too much and I definitely agree with that. It just adds a lot of boilerplate.

**Review by Sheepyhead — APPROVED** [review]

**Comment by nicopap** [comment]
I think the `winit` line is fine. It's actually quite useful for the user. If they already know `winit`, they know they can re-use their knowledge in bevy thanks to that doc line. And if they don't know `winit`, but need more in depth understanding of bevy's internal, it's great that they have some hint of where to go. I've reconsidered my position.

I mean that "It is available to the end user" is a bit redundant on a doc string. As a user, I don't really need to know that a type is available to me in a documentation page. If I am reading the documentation for that type, it means that necessarily it is available to me, so no need to tell it. Maybe just keeping "this can be used in game logic"?

"Store the data of a button" is a bit of an awkward sentence. And all things considered, doesn't add much that is not already said, I find the doc is improved with the sentence removed.

File: crates/bevy_input/src/mouse.rs
Comment (alice-i-cecile) [inline crates/bevy_input/src/mouse.rs:48]:
```suggestion
```

File: crates/bevy_input/src/mouse.rs
Comment (alice-i-cecile) [inline crates/bevy_input/src/mouse.rs:78]:
```suggestion
```

**Comment by alice-i-cecile** [comment]
Happy to merge this once @nicopap approves :)

**Review by nicopap — APPROVED** [review]

**Comment by nicopap** [comment]
Ah, this github ui confuses me. It's all good for me now

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**
